mautic/mautic
# Support for date format modifier to tokens

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/286
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/6225
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Add new modifiers to contact fields to date format.  Use two params date and time from configurations as default formatting.
![image](https://user-images.githubusercontent.com/462477/41714540-f1326bb6-7550-11e8-9d6f-8204ff03d3bc.png)

`{contactfield=aliasofyourcustomdatefield|datetime}`
`{contactfield=aliasofyourcustomdatefield|date}`
`{contactfield=aliasofyourcustomdatefield|time}`


#### Steps to test this PR:
1.  Create custom contact date field and create contact and fill this custom field 
2.  Create email with these contact tokens:

`{contactfield=aliasofyourcustomdatefield}`
`{contactfield=aliasofyourcustomdatefield|datetime}`
`{contactfield=aliasofyourcustomdatefield|date}`
`{contactfield=aliasofyourcustomdatefield|time}`

3. Send email to contact and see if your dates displayed in you predefined formats
